community/modules/compute/gke-node-pool/gpu_definition.tf

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+## Required variables:
+#  guest_accelerator
+#  machine_type
+
+locals {
+  # example state; terraform will ignore diffs if last element of URL matches
+  # guest_accelerator = [
+  #   {
+  #     count = 1
+  #     type  = "https://www.googleapis.com/compute/beta/projects/PROJECT/zones/ZONE/acceleratorTypes/nvidia-tesla-a100"
+  #   },
+  # ]
+  accelerator_machines = {
```

Comment:
Is it possible to share this code with other modules (using the dup/diff pre-commit). I realize that the GKE data is probably more rich than needed for vm-instance or Slurm, would it be possible to have all use the more detailed data and then just dumb it down when injecting into other modules. Other options would be to omit the default part from this file and add it in main.tf. It is already a pain having to keep this list updated manually, and I would rather not have to do it in 2 places.

Reply:
Updated file and added to dup check